### PR TITLE
[CN-479] Use TLS for backup server sidecar

### PR DESCRIPTION
**Pull request by dzeromski-hazelcast**

### CODE REVIEW

File: controllers/hazelcast/hazelcast_resources.go [hunk 1]
```diff
@@ -1097,6 +1119,17 @@ func hostPathVolume(h *hazelcastv1alpha1.Hazelcast) v1.Volume {
 	}
 }
 
+func tlsVolume(h *hazelcastv1alpha1.Hazelcast) v1.Volume {
+	return v1.Volume{
+		Name: "tls-secret",
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName: "tls-secret",
```
Comment (mtyazici) [inline controllers/hazelcast/hazelcast_resources.go:1127]: Maybe we can use a name that will not collide with possible other secrets. Also would it be good to put this name under `naming` package?

File: api/v1alpha1/hazelcast_types.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ type HazelcastSpec struct {
 
 	// B&R Agent configurations
 	// +optional
-	// +kubebuilder:default:={repository: "docker.io/hazelcast/platform-operator-agent", version: "0.1.6"}
```
Comment (dzeromski-hazelcast) [inline api/v1alpha1/hazelcast_types.go:85]: Temp change for PR tests to pass before we release agent.

File: controllers/hazelcast/hazelcast_resources.go (hunk 1 above)
Comment (dzeromski-hazelcast) [reply controllers/hazelcast/hazelcast_resources.go]: Fixed.

File: api/v1alpha1/hazelcast_types.go (hunk 2 above)
Comment (dzeromski-hazelcast) [reply api/v1alpha1/hazelcast_types.go:100]: Fixed.

**Review by SeriyBg — APPROVED** [review]

**Review by mtyazici — APPROVED** [review]

**Pull request merged**